[Keycloak CI] - ResetPasswordTest.resetPasswordExpiredCode failures

### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

login/ui

### Describe the bug

Reference run: https://github.com/keycloak/keycloak/actions/runs/10611716253/attempts/1

```
--------- org.keycloak.testsuite.forms.ResetPasswordTest output end  ---------
Error:  Tests run: 40, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 101.577 s <<< FAILURE! - in org.keycloak.testsuite.forms.ResetPasswordTest
Error: OR] org.keycloak.testsuite.forms.ResetPasswordTest.resetPasswordExpiredCode  Time elapsed: 1.716 s  <<< FAILURE!
-- java.lang.AssertionError: Expected LoginPasswordResetPage but was Sign in to test (https://localhost:8543/auth/realms/test/protocol/openid-connect/auth?response_type=code&client_id=test-app&redirect_uri=https%3A%2F%2Flocalhost%3A8543%2Fauth%2Frealms%2Fmaster%2Fapp%2Fauth&state=e4053a15-100d-40ad-9bc4-4190cb3575bf&scope=openid)
-- 	at org.junit.Assert.fail(Assert.java:89)
-- 	at org.junit.Assert.assertTrue(Assert.java:42)
-- 	at org.keycloak.testsuite.pages.AbstractPage.assertCurrent(AbstractPage.java:47)
-- 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
-- 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
-- 	at org.jboss.arquillian.graphene.proxy.GrapheneProxyHandler.invokeReal(GrapheneProxyHandler.java:143)
-- 	at org.jboss.arquillian.graphene.proxy.GrapheneContextualHandler$1.invoke(GrapheneContextualHandler.java:180)
-- 	at org.jboss.arquillian.graphene.proxy.GrapheneContextualHandler$2.call(GrapheneContextualHandler.java:252)
-- 	at org.jboss.arquillian.graphene.context.BrowserActions.performAction(BrowserActions.java:62)
-- 	at org.jboss.arquillian.graphene.proxy.GrapheneContextualHandler.invoke(GrapheneContextualHandler.java:248)
-- 	at org.jboss.arquillian.graphene.proxy.GrapheneProxyHandler.intercept(GrapheneProxyHandler.java:106)
-- 	at org.jboss.arquillian.graphene.proxy.GrapheneProxyHandler.intercept(GrapheneProxyHandler.java:81)
-- 	at org.jboss.arquillian.graphene.proxy.GrapheneContextualHandler.intercept(GrapheneContextualHandler.java:92)
-- 	at org.keycloak.testsuite.pages.LoginPasswordResetPage_ByGraphene.assertCurrent(Unknown Source)
-- 	at org.keycloak.testsuite.forms.ResetPasswordTest.initiateResetPasswordFromResetPasswordPage(ResetPasswordTest.java:475)
-- 	at org.keycloak.testsuite.forms.ResetPasswordTest.resetPasswordExpiredCode(ResetPasswordTest.java:514)
-- 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
-- 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
-- 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
-- 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
-- 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
-- 	at org.jboss.arquillian.junit.Arquillian$8$1.invokeMethod(Arquillian.java:319)
-- 	at org.jboss.arquillian.junit.MethodInvoker$1.invoke(MethodInvoker.java:18)
-- 	at org.jboss.arquillian.container.test.impl.execution.LocalTestExecuter.execute(LocalTestExecuter.java:57)
-- 	at org.keycloak.testsuite.arquillian.ModelTestExecutor.execute(ModelTestExecutor.java:49)
-- 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
-- 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
-- 	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
-- 	at org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:103)
-- 	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:90)
-- 	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:133)
-- 	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:105)
-- 	at org.jboss.arquillian.core.impl.EventImpl.fire(EventImpl.java:62)
-- 	at org.jboss.arquillian.container.test.impl.execution.ClientTestExecuter.execute(ClientTestExecuter.java:50)
-- 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
-- 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
-- 	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
-- 	at org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:103)
-- 	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:90)
-- 	at org.jboss.arquillian.container.test.impl.client.ContainerEventController.createContext(ContainerEventController.java:128)
-- 	at org.jboss.arquillian.container.test.impl.client.ContainerEventController.createTestContext(ContainerEventController.java:118)
-- 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
-- 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
-- 	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
-- 	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
-- 	at org.jboss.arquillian.test.impl.TestContextHandler.createTestContext(TestContextHandler.java:116)
-- 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
-- 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
-- 	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
-- 	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
-- 	at org.jboss.arquillian.test.impl.TestContextHandler.createClassContext(TestContextHandler.java:83)
-- 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
-- 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
-- 	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
-- 	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
-- 	at org.jboss.arquillian.test.impl.TestContextHandler.createSuiteContext(TestContextHandler.java:69)
-- 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
-- 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
-- 	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
-- 	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
-- 	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:133)
-- 	at org.jboss.arquillian.test.impl.EventTestRunnerAdaptor.test(EventTestRunnerAdaptor.java:139)
-- 	at org.jboss.arquillian.junit.MethodInvoker.invoke(MethodInvoker.java:15)
-- 	at org.jboss.arquillian.junit.Arquillian$8.evaluate(Arquillian.java:326)
-- 	at org.jboss.arquillian.junit.Arquillian$4.evaluate(Arquillian.java:198)
-- 	at org.jboss.arquillian.junit.Arquillian.multiExecute(Arquillian.java:344)
-- 	at org.jboss.arquillian.junit.Arquillian.access$200(Arquillian.java:48)
-- 	at org.jboss.arquillian.junit.Arquillian$5.evaluate(Arquillian.java:209)
-- 	at org.keycloak.testsuite.AssertEvents$1.evaluate(AssertEvents.java:79)
-- 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
-- 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
-- 	at org.jboss.arquillian.junit.Arquillian$7$1.invoke(Arquillian.java:273)
-- 	at org.jboss.arquillian.container.test.impl.execution.ClientBeforeAfterLifecycleEventExecuter.execute(ClientBeforeAfterLifecycleEventExecuter.java:88)
-- 	at org.jboss.arquillian.container.test.impl.execution.ClientBeforeAfterLifecycleEventExecuter.on(ClientBeforeAfterLifecycleEventExecuter.java:66)
-- 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
-- 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
-- 	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
-- 	at org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:103)
-- 	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:90)
-- 	at org.jboss.arquillian.container.test.impl.client.ContainerEventController.createContext(ContainerEventController.java:128)
-- 	at org.jboss.arquillian.container.test.impl.client.ContainerEventController.createBeforeContext(ContainerEventController.java:114)
-- 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
-- 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
-- 	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
-- 	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
-- 	at org.jboss.arquillian.test.impl.TestContextHandler.createTestContext(TestContextHandler.java:116)
-- 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
-- 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
-- 	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
-- 	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
-- 	at org.jboss.arquillian.test.impl.TestContextHandler.createClassContext(TestContextHandler.java:83)
-- 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
-- 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
-- 	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
-- 	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
-- 	at org.jboss.arquillian.test.impl.TestContextHandler.createSuiteContext(TestContextHandler.java:69)
-- 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
-- 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
-- 	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
-- 	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
-- 	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:133)
-- 	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:105)
-- 	at org.jboss.arquillian.test.impl.EventTestRunnerAdaptor.fireCustomLifecycle(EventTestRunnerAdaptor.java:159)
-- 	at org.jboss.arquillian.junit.Arquillian$7.evaluate(Arquillian.java:267)
-- 	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
-- 	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
-- 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
-- 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
-- 	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
-- 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
-- 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
-- 	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
-- 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
-- 	at org.jboss.arquillian.junit.Arquillian$2.evaluate(Arquillian.java:160)
-- 	at org.jboss.arquillian.junit.Arquillian.multiExecute(Arquillian.java:344)
-- 	at org.jboss.arquillian.junit.Arquillian.access$200(Arquillian.java:48)
-- 	at org.jboss.arquillian.junit.Arquillian$3.evaluate(Arquillian.java:171)
-- 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
-- 	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
-- 	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
-- 	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
-- 	at org.jboss.arquillian.junit.Arquillian.run(Arquillian.java:109)
-- 	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:377)
-- 	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:284)
-- 	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:248)
-- 	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:167)
-- 	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:456)
-- 	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:169)
-- 	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:595)
-- 	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:581)
```

### Version

latest

### Regression

- [ ] The issue is a regression

### Expected behavior

Successful test

### Actual behavior

Failing test

### How to Reproduce?

Run multiple times / Run in GHA

### Anything else?

_No response_

--- a/testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/pages/LoginPage.java
+++ b/testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/pages/LoginPage.java
@@ -19,12 +19,14 @@ package org.keycloak.testsuite.pages;
 
 import org.jboss.arquillian.test.api.ArquillianResource;
 import org.junit.Assert;
+import org.keycloak.common.util.Retry;
 import org.keycloak.testsuite.util.DroneUtils;
 import org.keycloak.testsuite.util.OAuthClient;
 import org.keycloak.testsuite.util.WaitUtils;
 import org.openqa.selenium.By;
 import org.openqa.selenium.NoSuchElementException;
 import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
 import org.openqa.selenium.support.FindBy;
 
 import static org.keycloak.testsuite.util.UIUtils.clickLink;
@@ -233,7 +235,14 @@ public class LoginPage extends LanguageComboboxAwarePage {
     }
 
     public void resetPassword() {
-        clickLink(resetPasswordLink);
+        // Since Chrome 128, the user can be still kept on the "Login page" after click to "Forget Password" link. Clicking the "Forget Password" link another
+        // time usually helps. Limit to 4 attempts for now.
+        Retry.execute(() -> {
+            clickLink(resetPasswordLink);
+            if (driver instanceof ChromeDriver) {
+                Assert.assertEquals("Forgot Your Password?", PageUtils.getPageTitle(driver));
+            }
+        }, 4, 0);
     }
 
     public void setRememberMe(boolean enable) {

